airbyte-db/lib/src/main/java/io/airbyte/db/mongodb/MongoUtils.java

```diff
@@ -121,7 +121,12 @@ private static ObjectNode readDocument(final BsonReader reader, final ObjectNode
 
   private static void transformToStringIfMarked(final ObjectNode jsonNodes, final List<String> columnNames, final String fieldName) {
     if (columnNames.contains(fieldName + AIRBYTE_SUFFIX)) {
-      jsonNodes.put(fieldName, jsonNodes.get(fieldName).asText());
+      JsonNode data = jsonNodes.get(fieldName);
+      if(data != null){
+        jsonNodes.put(fieldName, data.asText());
+      } else {
+        LOGGER.error("Field list out of sync, Document doesn't contain field: {}", fieldName);
```

Comment:
Many thanks for your PR. Basically, it looks good but got a few quick questions if you don't mind. What would happen with sync (its status, data in destination etc) if it got a null instead of field's name? Just wonder wouldn't we get corrupted data eventually as the "fieldName" name seems to be a pretty important attribute for migration. Can I ask you please to provide more details about the issue you observed? What were your source and destination, full logs, versions, etc? It's very important to make sure that we took everything into account and nothing was missed. Many thank in advance!

Reply:
Airbyte version: 0.30.22-alpha
MongoDB plugin version: 0.1.6
I migrate data from MongoDB to Kafka (then to Druid). This sync process is success before, maybe there is a change in MongoDB document schema (I don't own the database), a field is removed. 
I don;t know it's better to ignore the missing field or put an empty value for the field. In my case, I need to update schema in Druid configuration, so that skip the missing field is better for me.
file:///home/fatguy/Downloads/airbyte_mongoDb.txt

Reply:
@vudangngoc, you can merge this without bump versions. I'm just working on new changes for mongoDB-v2 and I want to add your and my changes with one version.

Reply:
@andriikorotkov Could you please give me the branch name? I'll edit this pull request to that branch

Reply:
@vudangngoc - https://github.com/airbytehq/airbyte/pull/8161 - this is a draft of my pull request.

Reply:
@vudangngoc, I added your changes to my pull request and to the new version of the connector. You can see it here - https://github.com/airbytehq/airbyte/pull/8161. Thank you for your contribution. You can close this pull request.